src/llfs/storage_file_builder.test.cpp

```diff
@@ -115,45 +112,43 @@ class StorageFileBuilderTest : public ::testing::Test
       if (config_block == config_blocks.back()) {
         expected_next_offset = llfs::PackedConfigBlock::kNullFileOffset;
       } else {
-        expected_next_offset =
-            batt::round_up_bits(12, expected_page0_offset + 64 * expected_num_slots);
+        expected_next_offset = batt::round_up_bits(
+            12, expected_page0_offset + llfs::PackedPageDeviceConfig::kSize * expected_num_slots);
       }
 
-      // Verify the packed congig block
+      // Verify the packed config block
       //
-      if (!(p_config_block.magic == llfs::PackedConfigBlock::kMagic &&
-            p_config_block.version == llfs::make_version_u64(0, 1, 0) &&
-            p_config_block.prev_offset == expected_prev_offset &&
-            p_config_block.next_offset == expected_next_offset &&
-            p_config_block.slots.size() == expected_num_slots &&
-            p_config_block.crc64 == p_config_block.true_crc64())) {
-        return false;
-      }
+      ASSERT_EQ(p_config_block.magic, llfs::PackedConfigBlock::kMagic);
+      ASSERT_EQ(p_config_block.version, llfs::make_version_u64(0, 1, 0));
+      ASSERT_EQ(p_config_block.prev_offset, expected_prev_offset);
+      ASSERT_EQ(p_config_block.next_offset, expected_next_offset);
+      ASSERT_EQ(p_config_block.slots.size(), expected_num_slots);
+      ASSERT_EQ(p_config_block.crc64, p_config_block.true_crc64());
     }
-    return true;
   }
 
-  void write_rand_data(llfs::IoRingRawBlockFile& test_file, i64 write_offset, i64 rand_buffer_len)
+  // Writes random data to disk given the file, offset, and amount of data to write. Used to
+  // randomize data of a test file to verify repeated write and read testing.
+  //
+  void write_rand_data(llfs::IoRingRawBlockFile& test_file, i64 write_offset, i64 write_size)
   {
     constexpr i64 kBufferSize = 512;
-    i64 rand_data_len = batt::round_up_bits(9, rand_buffer_len);
+    i64 write_size_aligned = batt::round_up_bits(9, write_size);
     i64 offset_aligned = batt::round_up_bits(12, write_offset);
-    int rand_fd = open("/dev/random", O_RDONLY);
+    llfs::StatusOr<int> rand_fd = llfs::open_file_read_only("/dev/random");
+    ASSERT_TRUE(rand_fd.ok()) << BATT_INSPECT(rand_fd.status());
 
     std::vector<std::aligned_storage_t<kBufferSize, kBufferSize>> data_vector;
-    data_vector.resize(rand_data_len / kBufferSize);
+    data_vector.resize(write_size_aligned / kBufferSize);
     llfs::MutableBuffer read_buffer(data_vector.data(), data_vector.size() * kBufferSize);
 
-    for (u64 i = 0; i < data_vector.size(); i++) {
-      int count = ::read(rand_fd, read_buffer.data(), kBufferSize);
-      ASSERT_EQ(count, kBufferSize);
-      read_buffer += count;
-    }
+    int count = ::read(*rand_fd, read_buffer.data(), write_size);
```

Comment:
Seems we are still using an unnecessarily large amount of memory to perform this operation.  Will follow up ...

Reply:
Made some changes based off of what we discussed earlier today.
- no longer have a need for an std::vector, just using aligned storage to manage the random data
- Always read in the set buffer size of 16k
- Write as much data up to the buffer size or the remaining data to write, whichever is less, and iterate until the requested amount of data is written

Can increase the buffer size to be 64k or higher.
I considered using llfs::read_fd, but decided against it in the case that the data written needs to be less than the buffer size.